core/app/models/spree/payment_source.rb

```diff
@@ -6,7 +6,7 @@ class PaymentSource < Spree::Base
 
     belongs_to :payment_method, optional: true
 
-    has_many :payments, as: :source
+    has_many :payments, as: :source, dependent: :restrict_with_error
```

Comment:
This is interesting. Why you opted for `restrict_with_error` here, although there is no UI associated with payment sources (that I am aware of)? I think here we actually should raise an exception